src/app/services/error-handling.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { NotificationService } from './notification.service';
+import { Observable } from 'rxjs';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Injectable()
+export class ErrorHandlingService {
+
+  constructor(private notificationService: NotificationService) { }
+
+  public handleServerError<T>() {
+    return (error: HttpErrorResponse): Observable<T> => {
+      if (error.status == 400) {
```

Comment:
Does typescript switch-case statements? Might make this a little cleaner. Only a small-potatoes complaint; not something that will prevent a merge in my opinion.

Reply:
🤦‍♂ It does.... I've written them before. I'll fix that.

Reply:
Instead it would be better to use `if ([400, 401, ...].includes(error.error.type))` or `if (error.status >= 400 && ...)`